src/RewardsManager.sol

```
@@ -662,11 +662,23 @@ contract RewardsManager is IRewardsManager, ReentrancyGuard {
         address pool_,
         uint256[] memory indexes_
     ) internal returns (uint256 updatedRewards_) {
-        // get the current burn epoch from the given pool
+
+        // get the current and previous burn epoch
         uint256 curBurnEpoch = IPool(pool_).currentBurnEpoch();
+        uint256 prevBurnEpoch = curBurnEpoch != 0 ? curBurnEpoch - 1 : 0;
```

not sure we want to move this logic / 2 external calls out of 2nd if and execute it also when not needed (that is if `curBurnEpoch == 0`). Not a big fan of `return 0` either, I suggest to keep this logic in place in the 2nd else where already is

```Solidity
if (block.timestamp <= curBurnTime + UPDATE_PERIOD) {
....
}
```

why's an issue if `UpdateExchangeRates` is emitted in this case? the updated rewards will still be 0 so we know it's a noop

---

OK, I thought it was a misleading event emit... We emit a `UpdateExchangeRates` but we don't actually update any exchange rates in this case.

I'll revert changes and just remove the unused error

---

Yeah, I don't see any issue in keeping the event, is probably better to have it for debugging purposes instead silently do nothing